pandas/tests/test_expressions.py

```diff
@@ -443,3 +443,23 @@ def test_bool_ops_warn_on_arithmetic(self):
                     r = f(df, True)
                     e = fe(df, True)
                     tm.assert_frame_equal(r, e)
+
+    @pytest.mark.parametrize("test_input,expected", [
+        (DataFrame([[0, 1, 2, 'aa'], [0, 1, 2, 'aa'],
+                    [0, 1, 5, 'bb'], [0, 1, 5, 'bb'],
+                    [0, 1, 5, 'bb']], columns=['a', 'b', 'c', 'dtype']),
+         DataFrame([[False, False], [False, False],
+                    [False, False], [False, False],
+                    [False, False]], columns=['a', 'dtype'])),
+        (DataFrame([[0, 3, 2, 'aa'], [0, 4, 2, 'aa'],
+                    [0, 1, 1, 'bb'], [0, 1, 2, 'bb'],
+                    [0, 2, 9, 'bb'], ['cc', 4, 3, 1]],
+                   columns=['a', 'b', 'c', 'dtype']),
+         DataFrame([[False, False], [False, False],
+                   [False, False], [False, False],
+                   [False, False], [False, False]],
+                   columns=['a', 'dtype'])),
+    ])
+    def test_bool_ops_column_name_dtype(self, test_input, expected):
+        # GH 22383 - .ne fails if columns containing column name 'dtype'
+        assert_frame_equal(test_input.loc[:, ['a', 'dtype']].ne(test_input.loc[:, ['a', 'dtype']]), expected)
```

Comment:
use 
```
result = 
expected = 
assert_frame_equal(result, expected)
```

Reply:
@jreback 
if i follow your guide, i think code to be long. so, could i declare dataframe variable on the top?

example

_issued_frame = DataFrame([[0, 1, 2, 'aa'], [0, 1, 2, 'aa'],
                    [0, 1, 5, 'bb'], [0, 1, 5, 'bb'],
                    [0, 1, 5, 'bb']], columns=['a', 'b', 'c', 'dtype']

    @pytest.mark.parametrize("test_input,expected", [
        (_issued_frame.loc[:, ['a', 'dtype']].ne(_issued_frame.loc[:, ['a', 'dtype']]),
         DataFrame([[False, False], [False, False],
                    [False, False], [False, False],
                    [False, False]], columns=['a', 'dtype']))
    ])
    def test_bool_ops_column_name_dtype(self, test_input, expected):
        # GH 22383 - .ne fails if columns containing column name 'dtype'
        assert_frame_equal(test_input, expected)